javalib/src/main/scala/java/util/regex/IndicesBuilder.scala

```diff
@@ -73,31 +78,35 @@ private[regex] class GroupStartMapper private (pattern: String, flags: String,
           s"Original pattern '$pattern' with flags '$flags' did match however.")
     }
 
-    // Prepare a `groupStartMap` array with the correct length filled with -1
-    val len = groupCount + 1 // index 0 is not used
-    val groupStartMap = new js.Array[Int](len)
-    var i = 0
+    val start = index // by definition
+    val end = start + allMatchResult(0).get.length()
+
+    // Prepare an `indices` array with the correct length filled with undefined
+    val len = groupCount + 1 // index 0 is the whole match
+    val indices = new IndicesArray(len)
+    indices(0) = js.Tuple2(start, end)
+    var i = 1
     while (i != len) {
-      groupStartMap(i) = -1
+      indices(i) = js.undefined
       i += 1
     }
```

Comment:
Doesn't a `js.Array` contain `js.undefined` by default?

Reply:
Not really. [It creates an array of empty slots, not of slots with actual `undefined` values.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Array#parameters). That causes the array to become a sparse array, which is less efficient.

Reply:
I have added a comment about that.